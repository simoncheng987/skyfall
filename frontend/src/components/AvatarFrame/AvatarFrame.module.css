.frame {
  width: 150px;
  height: 150px;
  background-color: white;
  border-radius: 5px;
  stroke-width: 2px;
  border-style: solid;
  border-color: black;
  overflow: hidden;
}

.image {
  width: 100%;
  height: 100%;
}
