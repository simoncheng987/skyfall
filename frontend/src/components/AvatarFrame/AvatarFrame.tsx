import React from 'react';
import styles from './AvatarFrame.module.css';

interface AvatarFrameProps {
  imgSrc?: string;
  className?: string;
}

export default function AvatarFrame({ imgSrc, className }: AvatarFrameProps) {
  return (
    <div className={`${className} ${styles.frame}`}>
      {imgSrc && <img src={imgSrc} alt="player" className={styles.image} />}
    </div>
  );
}

AvatarFrame.defaultProps = {
  imgSrc: '',
  className: '',
};
