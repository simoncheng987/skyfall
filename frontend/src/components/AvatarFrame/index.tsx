import AvatarFrame from './AvatarFrame';

export default AvatarFrame;
