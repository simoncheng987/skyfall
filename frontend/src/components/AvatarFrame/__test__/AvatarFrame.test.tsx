import React from 'react';
import renderer from 'react-test-renderer';
import AvatarFrame from '../AvatarFrame';

describe('Avatar frame', () => {
  const className = 'testClassName';

  it('matches snapshot with no prop', () => {
    const tree = renderer.create(<AvatarFrame />).toTree();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with className prop', () => {
    const tree = renderer.create(<AvatarFrame className={className} />).toTree();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with className prop', () => {
    const tree = renderer.create(<AvatarFrame imgSrc="host.png" />).toTree();
    expect(tree).toMatchSnapshot();
  });
});
