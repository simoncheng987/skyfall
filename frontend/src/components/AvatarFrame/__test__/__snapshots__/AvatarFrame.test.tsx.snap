// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar frame matches snapshot with className prop 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "className": "testClassName",
    "imgSrc": "",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "className": "testClassName frame",
    },
    "rendered": Array [
      "",
    ],
    "type": "div",
  },
  "type": [Function],
}
`;

exports[`Avatar frame matches snapshot with className prop 2`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "className": "",
    "imgSrc": "host.png",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="player"
        className="image"
        src="host.png"
      />,
      "className": " frame",
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "alt": "player",
          "className": "image",
          "src": "host.png",
        },
        "rendered": Array [],
        "type": "img",
      },
    ],
    "type": "div",
  },
  "type": [Function],
}
`;

exports[`Avatar frame matches snapshot with no prop 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "className": "",
    "imgSrc": "",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "className": " frame",
    },
    "rendered": Array [
      "",
    ],
    "type": "div",
  },
  "type": [Function],
}
`;
